.pre-commit-config.yaml

```diff
@@ -85,6 +85,12 @@ repos:
     additional_dependencies:
     - tomli
 
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: v1.7.1
+  hooks:
+  - id: mypy
+    exclude: ^docs|^\.jupyter|^\.github
+
```

Comment:
```suggestion
```
Following [this comment](https://github.com/astropy/astropy/issues/15170#issuecomment-1870944481) in https://github.com/astropy/astropy/issues/15170, I will remove the pre-commit hook before merging. This PR would then only add the configuration file (plus a few minor edits here and there to get mypy to pass).  In a future PR, we can set up a tox environment and enable it in CI.

Reply:
So, this PR just adds the capability to run mypi locally during development, but doesn't turn it on in CI?

Reply:
Correct!  Adding it to CI will come soon, probably.